### ISSUE TITLE
javalib Inet6Address should format IPv6 numeric addresses using lower case letters.

### ISSUE THREAD

**Issue by LeeTibbert** [scala-native/scala-native#2911]
Scala JVM (sbt console) formats IPv6 addresses for display using lowercase letters.
For example, the IPv6 multicast address "ffff::42:42"

I believe the fix is to change
```  
private val hexCharacters = "0123456789ABCDEF"

```
to 
```
private val hexCharacters = "0123456789abcdef"
```
Now using lowercase letters for hexadecimal is just plain silly,
but that is the JVM standard.

The impetus for changing this comes from it causing failures in some new tests under development
which check the new results against the results expected from JVM.

It is always problematic to reply on a specific format when parsing output, arrgh!

**Comment by LeeTibbert** [comment]
I was concerned that I had introduced this defect in my recent, rather large & complicated, PR #2877.
So I went diving back into history.  This UPPERCASE variant has been around since at least August, 2017.

Now, I personally believe that uppercase hex digits are the way they were created and the Universe/Multiverse
intended from the beginning.  Maguscule (upper case) came more than a millennium before miniscule. 
However, it a javalib developer's burden to follow the Java spec and practice as closely as feasible.
